tests/unit/test_io_utils.py

```diff
@@ -0,0 +1,218 @@
+        # because we are giving the path to this current unit test
+        # file which should exist and is not empty.
+        current_file_name = Path(__file__).parent.absolute()
+        try:
+            io_utils.assert_file_is_not_empty(
+                file_path=current_file_name, file_description="Io Utils"
+            )
+        except Exception as exc:
+            assert False, f"Function should not have raised any exception {exc}"
+
+        # asserts that an exception is raised by the function,
+        # because we are giving it a fake file path which shouldn't exist
+        with pytest.raises(FileExistsError):
+            io_utils.assert_file_is_not_empty(
+                file_path="/fake/file/path", file_description="Io Utils"
+            ), "Provided a fake file path, function is fail"
+
+    def test_assert_file_is_not_empty(self, temp_dir_path):
+
+        # Create temp file path
+        empty_temp_file_path = Path(temp_dir_path + "/empty.text")
+        # Check temp does not exits
+        if not os.path.exists(empty_temp_file_path):
+            with open(empty_temp_file_path, "w"):  # Create temp file
+                pass
+
+        with pytest.raises(EOFError):
+            io_utils.assert_file_is_not_empty(
+                file_path=empty_temp_file_path, file_description="Io Utils"
+            ), "Provided a fake file path, function is fail"
+
+        # Delete temp file
+        os.remove(empty_temp_file_path)
+
+    @pytest.mark.parametrize(
+        "workflow_id, validity, assert_msg",
+        [
+            (
+                "7ef69ca-6l9-44489-8ed-fce2876312c",
+                False,
+                "Should return false if workflow id is not hexadecimal",
+            ),
+            (
+                "7ez69ca5-6l9h-4449-8ej1-mce28763712c",
+                False,
+                "Should return false if workflow id is not hexadecimal",
+            ),
+            (
+                "7ef69ca5-0a9a-4449-8ed1-fce28763712c",
+                True,
+                "Should return True with valid ID",
+            ),
+        ],
+    )
+    def test_is_workflow_id_valid(self, workflow_id, validity, assert_msg):
+
+        with pytest.raises(TypeError):
+            io_utils.is_workflow_id_valid(
+                workflow_id=None
+            ), "Should raise an error if empty string is given"
+
+        with pytest.raises(ValueError):
+            io_utils.is_workflow_id_valid(
+                workflow_id=""
+            ), "Should raise an error if empty string is given"
+
+        assert (
+            bool(io_utils.is_workflow_id_valid(workflow_id=workflow_id)) is validity
+        ), assert_msg
+
+    def test_pretty_print_json(self):
+
+        # String holds the expected printout from function in a string, which includes
+        # 4 space indentation and new line characters
+        testing_out = (
+            """{\n    "id": "4bf7ca9c-0b39-48fb-9af7-83e3e488f62b",\n"""
```

Comment:
This test seems correct but it's testing a pretty simple json.  You might want to test something with at least 1 level of nesting to show it works on a non-trivial example.